### ISSUE TITLE
Slow memory copy on Jetson Nano

### ISSUE THREAD

**Issue by kmfrick** [tensorflow/tensorflow#39486]
<em>Please make sure that this is an issue related to performance of TensorFlow.
As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:performance_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Yes
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): https://forums.developer.nvidia.com/t/official-tensorflow-for-jetson-nano/71770/13
- TensorFlow version (use command below): 2.1.0
- Python version: 3.6.9
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version: CUDA 10.0.326 and cuDNN 7.6.3.28
- GPU model and memory: Maxwell 128-core GPU, Maximum Operating Frequency: 921MHz, 4ch x 16-bit LPDDR4
You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with:
1. TF 1.0: `python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"`
2. TF 2.0: `python -c "import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**
I'm running some experiments with semantic segmentation models on a Nano. One of the models I am using is [DeepLabv3 with a MNv2 backbone, trained on the ADE20K dataset](http://download.tensorflow.org/models/deeplabv3_mnv2_ade20k_train_2018_12_03.tar.gz). 

I noticed that the performance I was getting is very poor (~720ms per inference), on par with what I get on my five-year-old laptop running TensorFlow on CPU.

I debugged the issue and traced it to the memory copy, that is, `EagerTensor.numpy()`. Inference is very fast, as expected, but then converting the results to a format I can use in the rest of my workflow takes MUCH longer (up to 20x inference time if I use heavier networks like PSPNet).

I tried using TF1.x semantics with `tf.Session()` but no dice, `tf.Session.run()` becomes the bottleneck with similar timings. I tried converting the model to TensorRT... which makes the model even slower! (~920ms per inference)

Now, normally I would just say "oh well, memory copy is slow, duh" and try to find another way around it. But isn't the Nano supposed to share memory between its CPU and GPU? Is there any way to exploit this peculiarity? Or is it something I am doing wrong?

**Describe the expected behavior**
Total inference time including memory copy should be much lower.

**Standalone code to reproduce the issue**
Download DeepLabv3 as linked above, convert to SavedModel using [this script](https://github.com/kmfrick/TFLite_DeepLabv3_Inference/blob/master/convert_to_saved.py), then run
```python
import tensorflow as tf
import numpy as np
import time
imported = tf.saved_model.load('deeplabv3_mnv2_ade20k_train_2018_12_03_saved')
inp = np.random.uniform(0, 255, [1, 513, 513, 3])
t0 = time.time(); imported.signatures['serving_default'](tf.cast(inp, tf.uint8)); t1 = time.time(); print((t1 - t0) * 1000)
```

**Comment by tilakrayal** [comment]
@kmfrick ,
Is this still an issue?
Could you please update TensorFlow to the latest stable version v.2.7 and let us know if you are facing the same error. Thanks!

**Comment by kmfrick** [comment]
> @kmfrick , Is this still an issue? Could you please update TensorFlow to the latest stable version v.2.7 and let us know if you are facing the same error. Thanks!

I won't really have time to test it before the middle of January, but I will do so and report back asap. Has a shared memory allocator been implemented in TF 2.7?
